app/components/channel_link/channel_link.js

```diff
@@ -39,12 +50,33 @@ export default class ChannelLink extends React.PureComponent {
         return null;
     }
 
-    handlePress = () => {
-        this.props.actions.setChannelDisplayName(this.state.channel.display_name);
-        this.props.actions.handleSelectChannel(this.state.channel.id);
+    handlePress = async () => {
+        let {channel} = this.state;
 
-        if (this.props.onChannelLinkPress) {
-            this.props.onChannelLinkPress(this.state.channel);
+        if (!channel.id && channel.display_name) {
```

Comment:
channel can be null here according to line 73 or line 73 is not required

Reply:
Okay, I've updated the logic. It's not the channel but the `channel.id` can be null.  That's in a case of failed to join a channel where channel has `display_name` field only - coming from `post.props.channel_mentions`. I've updated L73 (now L76) to check `channel.id` instead.